exchanges/okx/okx_websocket.go

```diff
@@ -358,6 +358,12 @@ func (ok *Okx) Unsubscribe(channelsToUnsubscribe []stream.ChannelSubscription) e
 // handleSubscription sends a subscription and unsubscription information thought the websocket endpoint.
 // as of the okx, exchange this endpoint sends subscription and unsubscription messages but with a list of json objects.
 func (ok *Okx) handleSubscription(operation string, subscriptions []stream.ChannelSubscription) error {
+	if operationSubscribe == operation && len(subscriptions) > ok.Features.Supports.WebsocketMaxSubscriptionsPerConnectionSupported {
```

Comment:
I think you should be putting a `MaximumSubscriptions` property for a websocket connection in the `stream` package somewhere passed in via `SetupNewConnection`, and handling it there. It will be able to calculate if the desired subscriptions puts it over the limit.

This becomes a more universal solution instead of needing to add exchange level code like you've done in here because exchanges like Binance will simply terminate your connection if you try to subscribe to too many endpoints per connection.